ui/app/src/services/content.ts

```diff
@@ -807,23 +812,14 @@ export function getChildrenByPath(
   );
 }
 
-export function fetchItemsByPath(
-  site: string,
-  paths: string[],
-  options?: {
-    skipHomePathOverride?: boolean;
-  }
-): Observable<DetailedItem[]> {
-  const requests: Observable<DetailedItem>[] = [];
-  paths.forEach((path) =>
-    requests.push(
-      getDetailedItem(
-        site,
-        path === '/site/website' ? (options?.skipHomePathOverride ? path : '/site/website/index.xml') : path
-      )
+export function fetchItemsByPath(siteId: string, paths: string[], preferContent = true): Observable<DetailedItem[]> {
+  const qs = toQueryString({ siteId, paths, preferContent });
+  return get(`/studio/api/2/content/sandbox_items_by_path${qs}`).pipe(
+    pluck('response', 'items'),
+    map((items) =>
+      items.map((item) => ({ ...parseSandBoxItemToDetailedItem(item), stateMap: createItemStateMap(item.state) }))
```

Comment:
Add options.castAsDetailedItem and only apply this if such option is true. Try to create the function overloads to specify when it returns Detailed or Sandbox